### PR TITLE
[PySpark] - Allow spark session range

### PR DESCRIPTION
**Pull request by mariotaddeucci**

### CODE REVIEW

File: tools/pythonpkg/duckdb/experimental/spark/sql/session.py [hunk 1]
```diff
@@ -191,9 +190,20 @@ def newSession(self) -> "SparkSession":
...
+        start: int,
+        end: Optional[int] = None,
+        step: int = 1,
+        numPartitions: Optional[int] = None,
     ) -> "DataFrame":
-        raise ContributionsAcceptedError
+        if numPartitions:
+            raise ContributionsAcceptedError
+
+        if end is None and start >= 0:
```
Comment (Tishj) [inline tools/pythonpkg/duckdb/experimental/spark/sql/session.py:202]: There are no tests that don't trigger this condition
What happens if start is negative

Comment (Tishj) [reply tools/pythonpkg/duckdb/experimental/spark/sql/session.py:202]: Is there a reason we don't just make a function call to the `range` function of duckdb?
through `DuckDBPyConnection.table_function` perhaps?

Comment (mariotaddeucci) [reply tools/pythonpkg/duckdb/experimental/spark/sql/session.py]: Nice, I didn’t know about this table function, switched to it, thanks o/

**Review by Tishj — APPROVED** [review]
Thanks, looks good to me

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**